packages/cli/src/models/local/UnpackController.ts

```diff
@@ -0,0 +1,72 @@
+import path from 'path';
+import request from 'request-promise-native';
+import fs from 'fs-extra';
+
+import { Logger } from 'zos-lib';
+
+import stdout from '../../utils/stdout';
+import patch from '../../utils/patch';
+import cp from '../../utils/cp';
+
+const ora = patch('ora');
+const simpleGit = patch('simple-git/promise');
+
+interface KitConfig {
+  ignore: string[],
+  hooks: Object,
+}
+
+export default class UnpackController {
+  public async unpack(url: string, workingDirPath: string = '', config: KitConfig): Promise<void | never> {
+    if (!url) throw Error('A url must be provided.');
+    if (!config) throw Error('A config must be provided.');
+
+    const { exec } = cp;
+    const { readdir, remove, pathExists } = fs;
+
+    // because zos always spawns '.zos.lock' file
+    if ((await readdir(workingDirPath)).length > 1) throw Error('The directory must be empty');
+
+    try {
+      let spinner = ora(`Downloading kit from ${url}`).start();
+      const git = simpleGit(workingDirPath);
+      await git.init();
+      await git.addRemote('origin', url);
+      await git.pull('origin', 'stable');
```

Comment:
What do you think about changing the `pull` command for a `fetch` followed by a [sparse checkout](https://stackoverflow.com/a/33934496/12791), using the ignore paths, so we don't need to delete anything afterwards?

Reply:
Alternatively, we could replace the `ignore` field of the `kit.json` by a `files` property, so the kit owner must manually list which files or folders are part of the kit (as we do in both zos and zos-lib `package.json`s). In that case, the `checkout` would be much cleaner, since we can just call checkout with the list of files and folders to unpack.

```
git checkout [<tree-ish>] [--] <pathspec>...
           Overwrite paths in the working tree by replacing with the contents in the index or in
           the <tree-ish> (most often a commit).
```

I think this option is cleaner and safer than the sparse checkout, or the manual remove. WDYT?

Reply:
Was thinking about this for quite a while. I suggest yet another option. I want to reduce entrance effort for kit developers to minimum. What is why I don't think it is that good to ask them to either create a list of include/exclude files because it is time consuming and error prone.
Also, all extra files still have to be downloaded. I am thinking, since we have `stable` branch only for releases, what if we ask kit's developers to remove all extra files from that branch? Essentially implementing it on a git/branch level? That way files can't be offsync and don't need to be downloaded?
WDYT?

Reply:
If I understand correctly, it'd something like a `gh-pages` branch, but specific for the zepkit publishing, right? If we go that way, then I would not use the `stable` branch: it already has semantic in the context of git-flow, and it should not ignore files that are tracked under `master` or `develop`. We should ask them to create a dedicated branch, such as `zos-kit`, but that could add more friction than the whitelist.

How about making the `files` property in the `kit.json` optional? If it is not present, then everything is included; otherwise, it is passed as an argument to `checkout` so that only the chosen files are downloaded. I don't think it's much of a hassle if all files are downloaded by default.

Reply:
Love it. No downloads and zero friction.